plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -340,7 +340,7 @@ The default behaviour is to detect which access style to use based on the config
             storageClass, pathStyleAccess);
 
         AmazonS3 client = s3Service.client(metadata.settings(), maxRetries, useThrottleRetries, pathStyleAccess);
-        blobStore = new S3BlobStore(settings, client, bucket, serverSideEncryption, bufferSize, maxRetries, cannedACL, storageClass);
```

Comment:
`maxRetries` is now just used for logging above, which I don't think is necessary, or should not be done here. I think reading that setting can be moved down into building the client?